external/kafka/src/test/scala/org/apache/spark/streaming/kafka/DirectKafkaStreamSuite.scala

```diff
@@ -364,8 +365,8 @@ class DirectKafkaStreamSuite
 
     val batchIntervalMilliseconds = 100
     val estimator = new ConstantEstimator(100)
-    val messageKeys = (1 to 200).map(_.toString)
-    val messages = messageKeys.map((_, 1)).toMap
+    val messages = Map("foo" -> 200)
+    kafkaTestUtils.sendMessages(topic, messages)
```

Comment:
Not sure if you're just talking about for testing, or for the main code.
For production use the complication is that how messages are partitioned
into kafka is configurable at the time you're producing the messages, so
configuration of spark partitioner would have to match.

On Thu, Dec 10, 2015 at 2:14 PM, Holden Karau <notifications@github.com>
wrote:

> In
> external/kafka/src/test/scala/org/apache/spark/streaming/kafka/DirectKafkaStreamSuite.scala
> <https://github.com/apache/spark/pull/10089#discussion_r47278574>:
>
> > @@ -364,8 +365,8 @@ class DirectKafkaStreamSuite
> >
> >      val batchIntervalMilliseconds = 100
> >      val estimator = new ConstantEstimator(100)
> > -    val messageKeys = (1 to 200).map(_.toString)
> > -    val messages = messageKeys.map((_, 1)).toMap
> > +    val messages = Map("foo" -> 200)
> > +    kafkaTestUtils.sendMessages(topic, messages)
>
> While I haven't done much work with Kafka, it seems like we could maybe
> explicitly specify the partitioner for producer in the
> producerConfiguration to be round robin if we wanted to (although that
> requires some custom code from what I can tell) or rotating the partition
> key.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/10089/files#r47278574>.
>